### DISCUSSION TITLE
unnecessary  line here?

### DISCUSSION THREAD

**Discussion by jiahuat** [prometheus/pushgateway#464] [General]
commit: c2b05e61cc0065eaf9cf67f0ae7b669f90ff88a1
code at : storage/diskmetricstore.go line 246:
`checkPersist() // In case something has been written in the meantime.`

at this moment, `lastPersist` is updated, and the `lastWrite` is still unchanged,   the value of  `lastWrite.After(lastPersist)`  is always false,  so the checkPersist doesn't do work.

Or, Am I misunderstanding something?

**Comment by beorn7** [comment]
`lastPersist` is updated with something coming from a different goroutine (`time.AfterFunc` runs the function in its own goroutine). This might be older than the last update of `lastWrite`.
